src/test/java/org/terasology/launcher/StringIteratorInputStream.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher;
+
+import com.google.common.primitives.Ints;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * An {@code InputStream} fed by an {@code Iterator<String>}.
+ * <p>
+ * Testing something that consumes an {@link InputStream} {@linkplain BufferedReader#lines() line-by-line}? Your test
+ * fixture has a list of strings to use as inputs; how hard could it be to get an InputStream that will pass them one
+ * at a time?
+ * <p>
+ * About this hard, it turns out.
+ * <p>
+ * This was <strong>not</strong> written with high-volume or high-throughput use cases in mind. It will handle a few
+ * lines (or even a few kilobytes) in your test suite just fine. I do not recommend you run your server
+ * traffic through it.
+ */
+public class StringIteratorInputStream extends InputStream {
```

Comment:
I did not look at this, and will just (git) blame you in case something breaks here 😅

Reply:
This isn't your idea of fun???

Fair. The trick will be that you probably won't see an exception _here,_ it'll be when you upgrade to Java 17 or 18 or whatever and the implementation of BufferedReader.lines() changes somehow and some test hangs or fails for some invisible reason.

Not looking forward to it, tbh.